demo/full/scripts/controllers/ControlBar.jsx

```diff
@@ -1,74 +1,88 @@
-    positionElement = <LivePosition />;
-  } else {
-    positionElement = <PositionInfos
-      position={currentTime}
-      duration={duration}
-    />;
+  onClickSettings() {
+    const { changeDisplay } = this.props;
+    changeDisplay();
   }
 
-  return (
-    <div className="controls-bar-container">
-      { (!displayProgressBar) ?
-        null : <Progressbar player={player} />
-      }
-      <div className="controls-bar">
-        <PlayPauseButton
-          className={"control-button"}
-          player={player}
-        />
-        { positionElement }
-        <div className="controls-right-side">
-          <div className="volume">
-            <VolumeButton
-              className="control-button"
-              player={player}
-            />
-            <VolumeBar
-              className="control-button"
-              player={player}
-            />
-          </div>
-          <FullscreenButton
+  render() {
+    const {
+      player,
+      videoElement,
+      isContentLoaded,
+      isLive,
+      currentTime,
+      duration,
+    } = this.props;
+    const displayProgressBar = isContentLoaded;
+
+    let positionElement;
+    if (!displayProgressBar) {
+      positionElement = null;
+    } else if (isLive) {
+      positionElement = <LivePosition />;
+    } else {
+      positionElement = <PositionInfos
+        position={currentTime}
+        duration={duration}
+      />;
+    }
+
+    return (
+      <div className="controls-bar-container">
+        { (!displayProgressBar) ?
+          null : <Progressbar player={player} />
+        }
+        <div className="controls-bar">
+          <PlayPauseButton
             className={"control-button"}
             player={player}
-            videoElement={videoElement}
           />
+          { positionElement }
+          <div className="controls-right-side">
+            <div>
+              <Button
+                disabled={!isContentLoaded}
+                className='control-button'
+                onClick={
+                  () => {
+                    this.onClickSettings();
+                  }
+                }
+                value={String.fromCharCode(0xf013)}
+              />
+            </div>
+            <div className="volume">
+              <VolumeButton
+                className="control-button"
+                player={player}
+              />
+              <VolumeBar
+                className="control-button"
+                player={player}
+              />
+            </div>
+            <FullscreenButton
+              className={"control-button"}
+              player={player}
+              videoElement={videoElement}
+            />
+          </div>
         </div>
       </div>
-    </div>
-  );
-};
-
-export default withModulesState({
-  player: {
-    currentTime: "currentTime",
-    duration: "duration",
-    isContentLoaded: "isContentLoaded",
-    isLive: "isLive",
-  },
-})(ControlBar);
+    );
+  }
+};
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this empty statement. ([javascript:S1116](https://sonarcloud.io/organizations/rx-player/rules#rule_key=javascript%3AS1116))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=312&issues=AWXOEfDYlgbYnohY_20U&open=AWXOEfDYlgbYnohY_20U)

[ISSUE_KEY]: AWXOEfDYlgbYnohY_20U